flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ProgramDeployer.java

```diff
@@ -73,7 +86,7 @@ public JobExecutionResult fetchExecutionResult() {
 	 * Deploys a job. Depending on the deployment creates a new job cluster. It saves the cluster id in
 	 * the result and blocks until job completion.
 	 */
-	private <T> JobExecutionResult deployJob(ExecutionContext<T> context, JobGraph jobGraph, DynamicResult<T> result) {
+	private <T> void deployJob(ExecutionContext<T> context, JobGraph jobGraph, Result<T> result) {
```

Comment:
split this method? Extract `if (context.getClusterId() == null)` and `else` branches to separate methods?